### DISCUSSION TITLE
Merge upstream repository with my fork

### DISCUSSION THREAD

**Discussion by learnjavawithreshma** [community/community#23694] [New to GitHub]
<p>Hello,<br>
I’m totally new to GitHub. I have just started working on a project where the client asked me to fork their repository and work on it.  I downloaded the Github desktop app and managed to do this.</p>
<p>Now, the client found out that my fork does not have the latest changes from their main branch, so they asked me to merge the main branch with my fork.</p>
<p>I have been looking around for how to to this and I found these two articles:</p>
<p><a href="https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/merging-an-upstream-repository-into-your-fork" rel="noopener nofollow ugc">https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/merging-an-upstream-repository-into-your-fork</a></p>
<p><a href="https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/syncing-a-fork" rel="noopener nofollow ugc">https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/syncing-a-fork</a></p>
<p>I’m really confused now and have several questions:</p>
<ol>
<li>Which of the above articles do I need to follow</li>
<li>What is the difference between the two (That is merging an upstream repository and syncing a fork)</li>
<li>Both the articles above suggest using the git bash. Is there some way to achieve what I want via the GitHub desktop app?</li>
</ol>
<p>Any help would be greatly appreciated!</p>
<p>Thanks in advance!</p>

**Comment by otter-computer** [comment, marked as answer]
<p>Hi <a class="mention" href="/learnjavawithreshma">@learnjavawithreshma</a>, welcome to the GitHub Support Community!</p>
<p>Both of these help articles actually explain how to do mostly the same thing: merge an upstream branch into your own local copy. They just use slightly different commands to do so.</p>
<p>The “Syncing a fork” article is more for beginners, including adding the parent of your fork as a new remote and specifically targeting the <code>master</code> branch and giving you commands to use to sync that. The other article assumes you already have knowledge of these things and doesn’t explain as much, but it does do pretty much the same thing!</p>
<p>As for GitHub Desktop, I don’t have an article to share directly but Desktop should have some tools to help you with this. When you clone a fork it should give you some options on how you’re planning to use this fork, so I’d recommend giving these choices a read and select the best option for your project:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/6/a/6a3527254f45338063463c2f608ae52c19c0ce08.jpeg" data-download-href="/uploads/short-url/f9yutalrbhj9YBnr2Qdz0SZ9f6g.jpeg?dl=1" title="Screenshot 2020-10-22 at 16.09.39"><img src="https://user-images.githubusercontent.com/2439407/181078781-4039a2ee-f718-495d-a1a9-394e6f480762.jpeg" alt="Screenshot 2020-10-22 at 16.09.39" data-base62-sha1="f9yutalrbhj9YBnr2Qdz0SZ9f6g" width="690" height="496" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/6/a/6a3527254f45338063463c2f608ae52c19c0ce08_2_690x496.jpeg, https://aws1.discourse-cdn.com/github/optimized/3X/6/a/6a3527254f45338063463c2f608ae52c19c0ce08_2_1035x744.jpeg 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/6/a/6a3527254f45338063463c2f608ae52c19c0ce08_2_1380x992.jpeg 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/6/a/6a3527254f45338063463c2f608ae52c19c0ce08_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot 2020-10-22 at 16.09.39</span><span class="informations">2144×1544 437 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>As for actually syncing your fork, from the branch selector dropdown you can choose a branch to merge into your currently checked-out branch using the button at the bottom:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/5/9/597b3f66712ae802f676c750de3097349cddc11a.jpeg" data-download-href="/uploads/short-url/cLAycX9oNsEViq8wwVJIMVZf3V8.jpeg?dl=1" title="Screenshot 2020-10-22 at 16.10.26"><img src="https://user-images.githubusercontent.com/2439407/181078782-11088f65-dc65-4d5f-9cfd-56e3b9054a6e.jpeg" alt="Screenshot 2020-10-22 at 16.10.26" data-base62-sha1="cLAycX9oNsEViq8wwVJIMVZf3V8" width="690" height="496" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/5/9/597b3f66712ae802f676c750de3097349cddc11a_2_690x496.jpeg, https://aws1.discourse-cdn.com/github/optimized/3X/5/9/597b3f66712ae802f676c750de3097349cddc11a_2_1035x744.jpeg 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/5/9/597b3f66712ae802f676c750de3097349cddc11a_2_1380x992.jpeg 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/5/9/597b3f66712ae802f676c750de3097349cddc11a_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot 2020-10-22 at 16.10.26</span><span class="informations">2144×1544 277 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>You should see an option named <code>upstream/master</code> (or <code>main</code> or whatever default branch your project uses). If you select this option it will pull in any changes from the upstream parent repository allowing you to sync your fork with the upstream.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/a/c/ac9aaa817d9bb3b08edcb99ee3edc807c4e72928.jpeg" data-download-href="/uploads/short-url/oCVuIe0UvW25mCkXwI8PDsMpuju.jpeg?dl=1" title="Screenshot 2020-10-22 at 16.10.18"><img src="https://user-images.githubusercontent.com/2439407/181078783-d8668c29-9792-4516-bfc3-1fcf60478ecd.jpeg" alt="Screenshot 2020-10-22 at 16.10.18" data-base62-sha1="oCVuIe0UvW25mCkXwI8PDsMpuju" width="690" height="496" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/a/c/ac9aaa817d9bb3b08edcb99ee3edc807c4e72928_2_690x496.jpeg, https://aws1.discourse-cdn.com/github/optimized/3X/a/c/ac9aaa817d9bb3b08edcb99ee3edc807c4e72928_2_1035x744.jpeg 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/a/c/ac9aaa817d9bb3b08edcb99ee3edc807c4e72928_2_1380x992.jpeg 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/a/c/ac9aaa817d9bb3b08edcb99ee3edc807c4e72928_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot 2020-10-22 at 16.10.18</span><span class="informations">2144×1544 288 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Doing this should update your local repository—and therefore your own fork—with any changes made to the original parent.</p>

**Comment by learnjavawithreshma** [comment]
<p>Thank you so much for the detailed explanation. I will give it a try!</p>
<p>Regards,<br>
Reshma</p>
